arcgis-ios-sdk-samples/Layers/Group layers/LayersTableViewController.swift

```diff
@@ -0,0 +1,156 @@
+        headerView.layerVisibilitySwitch?.tag = section
+        headerView.delegate = self
+        
+        return headerView
+    }
+    
+    // MARK: - GroupLayersCellDelegate
+    
+    func didToggleSwitch(_ cell: GroupLayersCell, isOn: Bool) {
+        guard let indexPath = tableView.indexPath(for: cell) else { return }
+        if let groupLayer = getLayer(from: layers, at: indexPath.section) as? AGSGroupLayer, let childLayers = groupLayer.layers as? [AGSLayer], let childLayer = getLayer(from: childLayers, at: indexPath.row) {
+            childLayer.isVisible = isOn
+        }
+    }
+    
+    // MARK: - GroupLayersSectionViewDelegate
+    
+    func didToggleSwitch(_ sectionView: GroupLayersSectionView, section: Int, isOn: Bool) {
+        let layer = getLayer(from: layers, at: section)
+        layer?.isVisible = isOn
+        updateSwitchForRows(in: section, isEnabled: isOn)
+    }
+    
+    // MARK: - Helper methods
+    
+    /// Ensures we cannot toggle visibility of child layers when the parent is turned off.
+    ///
+    /// - Parameters:
+    ///     - section: Section number of the table view.
+    ///     - enabled: Indicates if the switch should be enabled or disabled.
+    func updateSwitchForRows(in section: Int, isEnabled: Bool) {
+        guard let visibleRows = tableView.indexPathsForVisibleRows else { return }
+        let visibleRowsForSection = visibleRows.filter { $0.section == section }
+        for indexPath in visibleRowsForSection {
+            if indexPath.section == section, let cell = tableView.cellForRow(at: indexPath) as? GroupLayersCell {
+                cell.layerVisibilitySwitch.isEnabled = isEnabled
+            }
+        }
+    }
+    
+    /// Modifies name of the layer.
+    ///
+    /// - Parameter name: Original name of the layer.
+    /// - Returns: A modified name or the original name of the layer.
+    func formattedValue(of name: String) -> String {
+        switch name {
+        case "DevA_Trees":
+            return "Dev A: Trees"
+        case "DevA_Pathways":
+            return "Dev A: Pathways"
+        case "DevA_BuildingShell_Textured":
+            return "Dev A: Building Shell Textured"
+        case "PlannedDemo_BuildingShell":
+            return "Planned Demo Building Shell"
+        case "DevelopmentProjectArea":
+            return "Development Project Area"
+        default:
+            return name
+        }
+    }
+    
+    /// To get layer from a list.
+    ///
+    /// - Parameters:
+    ///     - layers: List of layers.
+    ///     - section: Position of layer on the layers list.
+    /// - Returns: A layer if it exists else it returns nil.
+    func getLayer(from layers: [AGSLayer], at section: Int) -> AGSLayer? {
+        if section >= layers.count { return nil }
+        return layers[section]
+    }
+}
+
+class GroupLayersCell: UITableViewCell {
+    @IBOutlet var layerVisibilitySwitch: UISwitch!
```

Comment:
As you probably have noticed, the switches no longer have the proper padding on the right of the cell. You could solve that by constraining the trailing edge to the trailing edge of the margin of the superview (with a constant of zero), but there's also another problem. The label's text is very close to the switch. Too close, in my opinion. I think your first idea of setting the switch as the accessory view was a good one. We can't do that in the storyboard (at least with a prototype cell--it can be done with static cells), but we could do that here in code:

```swift
let layerVisibilitySwitch: UISwitch

required init?(coder aDecoder: NSCoder) {
    layerVisibilitySwitch = UISwitch()
    super.init(coder: aDecoder)
    accessoryView = layerVisibilitySwitch
}
```

![Screen Shot 2019-04-11 at 3 55 09 PM](https://user-images.githubusercontent.com/2257493/55998517-6ced5b80-5c73-11e9-9a21-599c0d39c05b.png)

The other option would be to use our own label in the storyboard so that we could add a minimum spacing constraint between our label and the switch.